pkg/server/testing/fake_execution_client.go

```diff
@@ -0,0 +1,315 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/services/execution"
+	"github.com/containerd/containerd/api/types/container"
+	google_protobuf "github.com/golang/protobuf/ptypes/empty"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+type calledDetail struct {
+	name     string
+	argument interface{}
+}
+
+var _ execution.ContainerService_EventsClient = &EventClient{}
+
+// EventClient is a test implementation of execution.ContainerService_EventsClient
+type EventClient struct {
+	Events chan *container.Event
+	grpc.ClientStream
+}
+
+// Recv is a test implementation of Recv
+func (cli *EventClient) Recv() (*container.Event, error) {
+	event := <-cli.Events
+	return event, nil
+}
+
+// FakeExecutionClient is a simple fake execution client, so that cri-containerd
+// can be run for testing without requiring a real containerd setup.
+type FakeExecutionClient struct {
+	sync.Mutex
+	called        []calledDetail
```

Comment:
Actually, we need assert functions to check ContainerList and called list. However, I'm fine with adding them in the future when we actually use them.